templates/master/00-master/_base/files/usr-local-bin-etcd-snapshot-restore-sh.yaml

```diff
@@ -17,17 +17,41 @@ contents:
     fi
 
     usage () {
-        echo 'Path to backup file and initial cluster are required: ./etcd-snapshot-restore.sh $path-to-backup $initial_cluster'
+        echo 'Path to backup file and initial cluster are required: ./etcd-snapshot-restore.sh <path-to-backup> $INITIAL_CLUSTER'
+        echo 'path-to-backup can be a directory containing kube static resources and snapshot db, or a single file containing snapshot db'
         exit 1
     }
 
     if [ "$1" == "" ] || [ "$2" == "" ]; then
         usage
     fi
 
-    BACKUP_FILE="$1"
     INITIAL_CLUSTER="$2"
     ASSET_DIR=./assets
+
+    RESTORE_STATIC_RESOURCES="true"
+    if [ -f "$1" ]; then
+      # For backward-compatibility, we support restoring from single snapshot.db file or single tar.gz file
+      if [[ "$1" =~ \.db$ ]]; then
+        RESTORE_STATIC_RESOURCES="false"
+        SNAPSHOT_FILE="$1"
+      elif [[ "$1" =~ \.tar\.gz$ ]]; then
+        BACKUP_FILE="$1"
+        tar xzf ${BACKUP_FILE} -C ${ASSET_DIR}/tmp/ snapshot.db
+        SNAPSHOT_FILE="${ASSET_DIR}/tmp/snapshot.db"
```

Comment:
Above four lines are added to support single tar.gz format (generated by 4.3.0).